lib/src/app.dart

```diff
@@ -22,9 +23,8 @@ class MyApp extends StatelessWidget {
     //
     // The ListenableBuilder Widget listens to the SettingsController for changes.
     // Whenever the user updates their settings, the MaterialApp is rebuilt.
-    return ListenableBuilder(
-      listenable: settingsController,
-      builder: (BuildContext context, Widget? child) {
+    return Watch(
```

Comment:
You dont want to put `Watch` at the top of the widget tree as it watches *all* the signals in it's subtree and rebuild the entire Material app whenever one of them changes. Use `mysignal.watch(context)` instead.

https://github.com/rodydavis/signals.dart/issues/29#issue-2017483958

Reply:
Thought that was a flaw of the `SignalWidget` and not `Watch`. But then, you are right. From the docs:

```dart
/// [Watch] is a drop-in replacement for [Builder] that will rebuild
/// when any signals inside the builder change
```

```dart
/// Rebuild the [Element] that the current signal is inside of
T watch(BuildContext context) => watchSignal<T>(context, this);
```

```dart
/// Watch a signal value and rebuild the context of the [Element]
/// if mounted and mark it as dirty
T watchSignal<T>(BuildContext context, ReadonlySignal<T> signal) {
```


Will update the sample. Thanks @jinyus

Reply:
Yeah, `SignalWidget` was just a wrapper around `Watch`